### PR TITLE
Revert signing code to fix cpp build

### PR DESCRIPTION
**Pull request by rohitnarula7176**
## Proposed changes

This PR reverts the signing code change I made which used SecKeyVerifySignature in my previous PR as it breaks cpp build. SecKeyVerifySignature was always using sha256 hashing and cpp was relying on this method to do sha1 hashing as well. 
The files NSData+JWT is the same code we had before which used SecKeyRawSign. Added these files back again unchanged. Deleting these files was breaking cpp build.
Also exposed the properties like, keyModulus and keyExponent of public key as it was before because cpp is relying on these properties being publicly exposed.

## Type of change

- [x] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [x] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: IdentityCore/src/cache/crypto/MSIDAssymetricKeyPair.m [hunk 1]
```diff
@@ -233,8 +234,15 @@ - (nullable NSData *)decrypt:(nonnull NSString *)encryptedMessageString
...
-    return [[NSString alloc] initWithData:signedData encoding:NSUTF8StringEncoding];
+    if ([message length] == 0)
+    {
+        MSID_LOG_WITH_CTX(MSIDLogLevelError, nil, @"Message to sign was empty");
+        return nil;
+    }
+    
+    NSData *hashedData = [[message dataUsingEncoding:NSUTF8StringEncoding] msidSHA256];
+    NSData *signedData = [hashedData msidSignHashWithPrivateKey:self.privateKeyRef];
+    return [signedData msidHexString];
```
Comment (oldalton) [inline IdentityCore/src/cache/crypto/MSIDAssymetricKeyPair.m:245]: Why does it need to be hex string? Will it potentially break other callers?

File: IdentityCore/src/util/NSData+JWT.m [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+#import "NSData+JWT.h"
+#import <CommonCrypto/CommonDigest.h>
+#import <Security/Security.h>
+#import <Security/SecKey.h>
+
+@implementation NSData (JWT)
+
+#if TARGET_OS_IPHONE
+
+- (NSData *)msidSignHashWithPrivateKey:(SecKeyRef)privateKey
```
Comment (oldalton) [inline IdentityCore/src/util/NSData+JWT.m:33]: Is this exact code that was here before? Just confirming.

**Review by oldalton — COMMENTED** [review]
Can you also fill in the PR template for quick reference?
Also, would appreciate if you update kSecAttrAccessible change as discussed offline last week :)
Thanks.

File: IdentityCore/src/util/NSData+JWT.m (hunk 2 above)
Comment (rohitnarula7176) [reply IdentityCore/src/util/NSData+JWT.m:33]: Yes this is the same exact code as before. MSAL cpp relies on this file NSData+JWT which I deleted and they were using this for sha1 hashing as well along with sha256 hashing so had to revert my change. The change I made in my previous PR will always perform SecKeyCreateSignature with sha256 hashing and will therefore break cpp code.

File: IdentityCore/src/cache/crypto/MSIDAssymetricKeyPair.m (hunk 1 above)
Comment (rohitnarula7176) [reply IdentityCore/src/cache/crypto/MSIDAssymetricKeyPair.m:245]: I wanted to discuss this with you as well. I think we should return a base64 string as signature cannot be shared over the internet in binary format.

Comment (rohitnarula7176) [reply IdentityCore/src/cache/crypto/MSIDAssymetricKeyPair.m:245]: Will update the code.

File: IdentityCore/src/util/MSIDJWTHelper.m [hunk 3]
```diff
@@ -74,6 +76,14 @@ + (NSString *)decryptJWT:(NSData *)jwtData
     return [[NSString alloc] initWithData:bits encoding:NSUTF8StringEncoding];
 }
 
++ (NSData *)sign:(SecKeyRef)privateKey
```
Comment (oldalton) [inline IdentityCore/src/util/MSIDJWTHelper.m:79]: Just curious where this is being used?

**Review by oldalton — APPROVED** [review]

File: IdentityCore/src/util/MSIDJWTHelper.m (hunk 3 above)
Comment (rohitnarula7176) [reply IdentityCore/src/util/MSIDJWTHelper.m:79]: This is used for signing the jwt with private key for at pop.

File: IdentityCore/src/cache/crypto/MSIDAssymetricKeyKeychainGenerator.m [hunk 4]
```diff
@@ -228,34 +236,42 @@ - (MSIDAssymetricKeyPair *)generateEphemeralKeyPair:(NSError **)error
 - (MSIDAssymetricKeyPair *)generateKeyPairForKeyDict:(NSDictionary *)attributes
```
[the remark this answers is not in the document]
Comment (rohitnarula7176) [reply IdentityCore/src/cache/crypto/MSIDAssymetricKeyKeychainGenerator.m:236]: Agree but will do it as a separate PR later.

**Pull request merged**